langchain/src/callbacks/base.ts

```diff
@@ -187,6 +188,40 @@ abstract class BaseCallbackHandlerMethodsClass {
     parentRunId?: string,
     tags?: string[]
   ): Promise<void> | void;
+
+  /**
+   * Called when embedding throws an error, after call to embedding.
+   * with the error and the run ID
+   */
+  handleEmbeddingError?(
+    err: Error,
+    runId: string,
+    parentRunId?: string
+  ): Promise<void> | void;
+
+  /**
+   * Called when embedding is started, before call to embedding.
+   * with the input text and the run ID
+   */
+  handleEmbeddingStart?(
+    embedding: Serialized,
+    texts: string[],
+    runId: string,
+    parentRunId?: string,
+    extraParams?: Record<string, unknown>,
+    tags?: string[],
+    metadata?: Record<string, unknown>
+  ): Promise<void> | void;
+
+  /**
+   * Called after embedding is completed, before it exits.
+   * with embeddings vector and the run ID
+   */
+  handleEmbeddingEnd?(
+    vector: number[],
```

Comment:
@nfcampos This should be `number[][]`, right? (can generate multiple embeddings per query)